modules/lang-painless/src/doc/java/org/elasticsearch/painless/JavadocExtractor.java

```diff
@@ -192,15 +202,103 @@ public int hashCode() {
     }
 
     public static class ParsedMethod {
-        public final String javadoc;
+        public final ParsedJavadoc javadoc;
         public final List<String> parameterNames;
 
-        public ParsedMethod(String javadoc, List<String> parameterNames) {
+        public ParsedMethod(ParsedJavadoc javadoc, List<String> parameterNames) {
             this.javadoc = javadoc;
             this.parameterNames = parameterNames;
         }
     }
 
+    public static ParsedJavadoc clean(Javadoc javadoc) {
+        JavadocDescription description = javadoc.getDescription();
+        List<JavadocBlockTag> tags = javadoc.getBlockTags();
+        List<String> cleaned = new ArrayList<>(description.getElements().size() + tags.size());
+        cleaned.addAll(stripInlineTags(description));
+        ParsedJavadoc parsed = new ParsedJavadoc(cleaned(cleaned));
+        for (JavadocBlockTag tag: tags) {
+            String tagName = tag.getTagName();
+            // https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#javadoctags
```

Comment:
Did you intend to link to Java 7 for this comment?

Reply:
Updated to java14.